### DISCUSSION TITLE
Actions in navigation bar

### DISCUSSION THREAD

**Discussion by Majkie** [filamentphp/filament#977] [Showcase]
Hi, so I've started working on a new function that I desperately needed in one of my projects. And the question is would you appreciate something like this in Filament admin?

<img width="1679" alt="Screenshot 2021-12-23 at 5 24 22" src="https://user-images.githubusercontent.com/37813691/147187669-025a4e85-ccbd-47b0-ba08-4f0bbfd1ba6d.png">

Basically, it would be possible to create same dropdown just by running something like `php artisan make:filament-action Profile --dropdown` and later define fields in ProfileAction (might need some ideas to change name for both actions and fields, to match exactly what they do).

```php
class ProfileAction extends DropdownAction
{
    public function getButton()
    {
        return IconButtonAction::make('Profile')
            ->icon('heroicon-o-user');
    }

    public function fields(): array
    {
        return [
            LabelField::make('Label')
                ->label('User profile'),

            LinkActionField::make('Profile')
                ->url('/admin/profile'),

            DividerField::make('Divider'),

            LinkActionField::make('Logout')
                ->action('logoutAction'),
        ];
    }

    public function logoutAction()
    {
        auth()->guard()->logout();

        return redirect()->route('welcome');
    }
}
```

The ProfileAction can be later registered the same way as Widgets, Pages and Resources (inside your config or just by having them in Filament/Actions folder).

These navigation actions can be defined as simple link/action or dropdown as well as your very custom actions.

**Comment by danharrin** [comment]
https://filamentphp.com/docs/4.x/navigation/user-menu
